rs/nervous_system/integration_tests/src/pocket_ic_helpers.rs

```diff
@@ -386,25 +402,30 @@ pub struct SnsTestCanisterIds {
 
 /// Function to allow directly installing and specifying the ids of the canisters
 /// which is only useful when we need to test something that is ID-specific
-pub fn install_sns_directly_with_snsw_versions(
+pub async fn install_sns_directly_with_snsw_versions(
     pocket_ic: &PocketIc,
     mut payloads: SnsCanisterInitPayloads,
     sns_canister_ids: Option<SnsTestCanisterIds>,
 ) -> SnsTestCanisterIds {
-    let create_canister = || -> Principal {
-        let id = pocket_ic.create_canister();
-        pocket_ic.add_cycles(id, STARTING_CYCLES_PER_CANISTER);
+    let create_canister = || async move {
```

Comment:
I like how it used to say `-> Principal`, especially because `id` is pretty ambiguous.

Reply:
I agree. The downside is that since the function is now async it would return a `impl Future<output=Principal>` which is way more verbose. Maybe there's another way to do it though.